app/services/listings/bump.rb

```diff
@@ -0,0 +1,38 @@
+module Listings
+  class Bump
+    def self.call(listing, user:)
+      new(listing, user: user).call
+    end
+
+    def initialize(listing, user:)
+      @listing = listing
+      @cost = listing.cost
+      @user = user
+      @org = Organization.find_by(id: listing.organization_id)
+    end
+
+    def call
+      purchaser = [@org, @user].detect { |who| who&.enough_credits?(@cost) }
```

Comment:
Reading note: we first see if the org will purchase, then the user.  This aligns with the past implementation in the controller.

Reply:
Yes, that's correct, thanks! Once this refactoring is "done" and things are less volatile I'll add Rubydocs to all the new classes. I was considering doing it right away but I don't want to rewrite them repeatedly.